src/restruct_poc/beams.hpp

```diff
@@ -0,0 +1,689 @@
+#pragma once
+
+#include <array>
+#include <numeric>
+
+#include <KokkosBlas.hpp>
+
+#include "beams_calcs.hpp"
+#include "interpolation.h"
+#include "types.hpp"
+
+#include "src/gebt_poc/quadrature.h"
+
+namespace oturb {
+
+//------------------------------------------------------------------------------
+// Beam input data structures
+//------------------------------------------------------------------------------
+
+struct BeamNode {
+    double s_;
+    std::array<double, 7> x_;
+    std::array<double, 7> q_;
+    std::array<double, 6> v_;
+    std::array<double, 6> a_;
```

Comment:
I would prefer to call these attributes by their full name, rather than the mathematical symbol used to represent them in the theory doc.

In addition, should try to remove using [magic numbers](https://stackoverflow.com/questions/47882/what-are-magic-numbers-and-why-do-some-consider-them-bad) if we can help it going forward. I have done this previously, but let's take this opportunity to buck the trend.

```
    std::array<double, kLieGroupComponents> position_;
    std::array<double, kLieGroupComponents> generalized_coords_; // or, displacements
    std::array<double, kLieAlgebraComponents> velocity_;
    std::array<double, kLieAlgebraComponents> acceleration_; // or, accln
```